utils/url.tsx

```diff
@@ -124,27 +129,35 @@ export function validateChannelUrl(url: string): React.ReactElement[] {
     const cleanedURL = cleanUpUrlable(url);
     const urlMatched = url.match(/[a-z0-9]([-_\w]*)[a-z0-9]/);
     if (cleanedURL !== url || !urlMatched || urlMatched[0] !== url || isDirectMessageFormat) {
-        if (url.length < 2) {
-            errors.push(formattedError(t('change_url.longer'), 'URLs must have at least 2 characters.'));
+        if (url.length < Constants.MIN_CHANNELNAME_LENGTH) {
+            errors.push(formattedError(t('change_url.longer'), 'URLs must have at least 2 characters.', formatMessages));
+        }
+
+        if (url.length > Constants.MAX_CHANNELNAME_LENGTH) {
```

Comment:
Looks like that the set of rules in the outermost `if` doesn't cover too long of channel names, so I think we need to add it because otherwise if a user has too long of a url but the channel display_name is ok, they're allowed to submit even though it is too long and the form submission will fail. Something like:
```ts
...
const tooLong = url.length > Constants.MAX_CHANNELNAME_LENGTH
if (cleanedURL !== url || !urlMatched || urlMatched[0] !== url || isDirectMessageFormat || tooLong) {
...
if (tooLong) {
  errors.push(formattedError(t('change_url.shorter'), 'U...
;
...
```